services/gitdiff/gitdiff.go

```diff
@@ -1278,13 +1279,25 @@ func readFileName(rd *strings.Reader) (string, bool) {
 	return name[2:], ambiguity
 }
 
-// GetDiffRangeWithWhitespaceBehavior builds a Diff between two commits of a repository.
+// DiffOptions represents the options for a DiffRange
+type DiffOptions struct {
+	BeforeCommitID     string
+	AfterCommitID      string
+	SkipTo             string
+	MaxLines           int
+	MaxLineCharacters  int
+	MaxFiles           int
+	WhitespaceBehavior string
+	DirectComparison   bool
+}
+
+// GetDiff builds a Diff between two commits of a repository.
 // Passing the empty string as beforeCommitID returns a diff from the parent commit.
 // The whitespaceBehavior is either an empty string or a git flag
-func GetDiffRangeWithWhitespaceBehavior(gitRepo *git.Repository, beforeCommitID, afterCommitID, skipTo string, maxLines, maxLineCharacters, maxFiles int, whitespaceBehavior string, directComparison bool) (*Diff, error) {
+func GetDiff(gitRepo *git.Repository, opts DiffOptions, files ...string) (*Diff, error) {
```

Comment:
`DiffOptions` is not particularly huge(89 bytes), but I do think that the diff function is a regularly used function for developers, so could it be done that the opts is passed as a pointer?

Reply:
Honestly pointers are not always the right answer. If you take a pointer you tell Go to consider if this needs to go to the heap.

Now in these cases the &DiffOptions here should be determinable as non-escaping so they should remain off the heap. If we really wanted to know we can use -m in the compilation stage to see if it escapes.

However, in the cases of the DiffOptions these are unchanged so Go should not even have to copy it even though its technically passed by value - but I don't know and decompiling this seems not worth it.

In either case the object is small and we throw enough crap at the heap that it barely matters.